### PR TITLE
feat(cli): improve renku run error messages

### LINKED ISSUE #2836: Improve `renku run` error messages
https://github.com/SwissDataScienceCenter/renku-python/blob/develop/renku/core/workflow/plan_factory.py#L329
--> Not that helpful. It should explain why this is an issue (Renku treats the whole directory as an output/as generated by this run, so it can't already contain files), and steps the user can take (delete the files, use `--param` to not treat it as a folder but a string instead or use a sub-subfolder).

https://github.com/SwissDataScienceCenter/renku-python/blob/develop/renku/command/run.py#L181
--> Mention why the command cannot be executed, it wasn't found, so $PATH might be wrong or it was misspelled.

### PR DESCRIPTION
**Pull request by Panaetius**
closes #2836

### CODE REVIEW

File: renku/core/errors.py [hunk 1]
```diff
@@ -212,28 +211,16 @@ class InvalidOutputPath(RenkuException):
...
-                "Once you have removed files that should be used as outputs,\n"
-                "you can safely rerun the previous command."
-                "\nYou can use --output flag to specify outputs explicitly."
-            )
-        else:
-            msg += "\n\nIf you want to track the command anyway use " "--no-output option."
+
+        msg = (
+            "There are not any detected outputs in the repository. This can be due to your command not creating "
+            "any new files or due to files that get created already existing before the command was run. In the "
```
Comment (mohammad-sdsc) [inline renku/core/errors.py:219]: I'm not sure about this: `already existing` -> `already exist`

File: renku/core/workflow/plan_factory.py [hunk 2]
```diff
@@ -79,19 +79,21 @@ def __init__(
...
-            raise errors.UsageError("Command line can not be empty.")
+            raise errors.UsageError("Command line can not be empty. Please specify a command to execute.")
 
         if not directory:
             directory = os.getcwd()
         self.directory = Path(directory)
         if not self.directory.exists():
-            raise errors.UsageError("Directory must exist.")
+            raise errors.UsageError(
+                "Current directory doesn't exist. Please make sure you are inside an existing directory."
```
Comment (mohammad-sdsc) [inline renku/core/workflow/plan_factory.py:89]: This message looks a bit weird to me. I'd change to `f"Directory doesn't exist: {self.directory}"

File: renku/core/workflow/plan_factory.py [hunk 3]
```diff
@@ -79,19 +79,21 @@ def __init__(
...
+            )
 
         if not working_dir:
             working_dir = os.getcwd()
         self.working_dir = Path(working_dir)
         if not self.working_dir.exists():
-            raise errors.UsageError("Working Directory must exist.")
+            raise errors.UsageError("Repository path doesn't exist. Make sure you are inside a Renku repository.")
```
Comment (mohammad-sdsc) [inline renku/core/workflow/plan_factory.py:96]: It's good to print the directory in the message: `f"Repository path doesn't exist: {self.working_dir} ..."

**Review by mohammad-sdsc — COMMENTED** [review]
Thanks Ralf! Please see my comments.

File: renku/core/errors.py (hunk 1 above)
Comment (Panaetius) [reply renku/core/errors.py:219]: I think plural is correct, `due to files [...] already existing`

File: renku/core/workflow/plan_factory.py (hunk 2 above)
Comment (Panaetius) [reply renku/core/workflow/plan_factory.py:89]: This part of the code is really misleading. `directory` gets set to `os.getcwd()` and `working_dir` gets set to `client.repository.path`. And it's not something a user can set.

The error above only happens if someone does `cd somedir && rm -r ../somedir && renku run ...`, I think.

**Pull request merged**